agent/reconciler/host_reconciler.go

```diff
@@ -92,6 +94,12 @@ func (r *HostReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctr
 
 func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
 	logger := ctrl.LoggerFrom(ctx)
+	if err := r.populateHostDetails(ctx, byoHost); err != nil {
+		logger.Error(err, "error getting host platform details")
+		r.Recorder.Eventf(byoHost, corev1.EventTypeWarning, "GetHostInfoFailed", err.Error())
+		return ctrl.Result{}, err
+	}
+
```

Comment:
Adding this inside reconcile loop doesn't feel right. Are there any strong opinions?

Reply:
This executes only once as the function checks for empty struct and then calls the os detector.

Reply:
Moved it back to registration.